subsys/nfc/lib/Kconfig

```diff
@@ -21,6 +21,97 @@ config NFCT_IRQ_PRIORITY
 	  Sets NFC interrupt priority.
 	  Levels are from 0 (highest priority) to 6 (lowest priority)
 
+config NFC_THREAD_CALLBACK
+	bool "Thread context for NFC callbacks"
+	select RING_BUFFER
+	default y
+	help
+	  Use this option to decouple the user NFC callback
+	  from the NFC interrupt context. When enabled, the user
+	  callback is called from a thread (System Workqueue
+	  or a dedicated thread depending on the CONFIG_NFC_OWN_THREAD
+	  setting) instead of being called from the interrupt context.
+	  Using this option ensures robust NFC communication
+	  (deterministic time of the interrupt handling) and allows
+	  deferred callback execution, while it results in higher memory
+	  footprint and CPU load.
+
+if NFC_THREAD_CALLBACK
+
+config HEAP_MEM_POOL_SIZE
+	default 512
+
+config NFC_RING_SIZE
+	int "Size of ring buffer in bytes."
+	default 2048
+	help
+	  Size of the ring buffer.
+	  The specified size is expressed as n bytes.
+
+config NFC_LIB_CTX_MAX_SIZE
+	int "Maximum size of NFC library context in bytes."
+	default 16
```

Comment:
Please add help for all public Kconfig options to render them nicely in the documentation